### DISCUSSION TITLE
Cant build SPA mode in production

### DISCUSSION THREAD

**Discussion by FragsterAt** [quasarframework/quasar#9555] [CLI - PWA mode] [answered]
![изображение](https://user-images.githubusercontent.com/20900647/120818916-b8ea7100-c55b-11eb-83c1-c8ac07be7a56.png)
Last quasar v2 beta, there is no section for simple SPA mode, so I post it in PWA section
When i run `npx quasar build` I get the following errors:

```
...
App • Cleaned build artifact: "C:\Users\users\Documents\GitHub\project\dist\spa"
 App •  WAIT  • Compiling of "UI" in progress...
 App •  DONE  • "UI" compiled with errors • 19903ms

 App •  ERROR  •  UI  

undefined


 App •  COMPILATION FAILED  • Please check the log above for details.
```

but dev server works, also debug build `npx quasar build -d` succeed. How can I fix prod build?

**Comment by hawkeye64** [comment, marked as answer]
@FragsterAt The error is in your code. But, try this `quasar build --debug` and see if it indicates better for you.

**Reply by FragsterAt** [reply]
If it error in code, why `npx quasar build -d` succeed? How can i find this error? Usual it write error line, so I can fix error, but not now. Also, it builds up to terser plugin (or even further) before error:
![изображение](https://user-images.githubusercontent.com/20900647/120825498-200b2400-c562-11eb-8061-a4924a740150.png)

**Reply by FragsterAt** [reply]
> But, try this quasar build --debug and see if it indicates better for you.

it causes no error, but I want to build release version.

**Reply by hawkeye64** [reply]
@FragsterAt Try and see if a linter will find it. I don't know how you installed your kit. Generally I do this

`yarn lint`

**Reply by FragsterAt** [reply]
This is an error with minifier: when I add `minify: false` in build section of quasar.conf it succeed. But I need minified bundle for prod.

**Reply by hawkeye64** [reply]
Sounds like terser is getting hung up on something. Plus, with the "undefined" above, this usually means you are using an undeclared variable. Try using a linter to find it.

**Reply by FragsterAt** [reply]
there was no eslint installed globally, so it cant run. After insataling it build succeed in all modes (and yes, there was no problems ^)). But error message is completely messing, it should be more clear
